### PR TITLE
Update docs on `gcs.service_account` usage

### PR DESCRIPTION
**Pull request by dimitarvdimitrov**
Signed-off-by: Dimitar Dimitrov <dimitar.dimitrov@grafana.com>

#### What this PR does

Clarify that the value of the parameter needs to be JSON, not a path to a file.

### CODE REVIEW

**Review by 56quarters — APPROVED** [review]
LGTM

**Review by treid314 — APPROVED** [review]
LGTM!

**Review by jdbaldry — APPROVED** [review]
LGTM

File: pkg/storage/bucket/gcs/config.go [hunk 1]
```diff
@@ -25,5 +25,5 @@ func (cfg *Config) RegisterFlags(f *flag.FlagSet) {
 // RegisterFlagsWithPrefix registers the flags for GCS storage with the provided prefix
 func (cfg *Config) RegisterFlagsWithPrefix(prefix string, f *flag.FlagSet) {
 	f.StringVar(&cfg.BucketName, prefix+"gcs.bucket-name", "", "GCS bucket name")
-	f.Var(&cfg.ServiceAccount, prefix+"gcs.service-account", "JSON representing either a Google Developers Console client_credentials.json file or a Google Developers service account key file. If empty, fallback to Google default logic.")
+	f.Var(&cfg.ServiceAccount, prefix+"gcs.service-account", "JSON either from a Google Developers Console client_credentials.json file, or a Google Developers service account key. Needs to be valid JSON, not a filesystem path. If empty, fallback to Google default logic.")
```
Comment (pracucci) [inline pkg/storage/bucket/gcs/config.go:28]: > If empty, fallback to Google default logic.

I think we need to clarify this. We should check what's the Google SDK default logic and explain it in plain text. I don't know what's that, so our users don't know neither.

Comment (dimitarvdimitrov) [reply pkg/storage/bucket/gcs/config.go]: pushed the change in https://github.com/grafana/mimir/pull/1690/commits/0ffa496476000f55ff60d40096775510c290b197, PTAL

File: pkg/storage/bucket/gcs/config.go [hunk 2]
```diff
@@ -25,5 +25,10 @@ func (cfg *Config) RegisterFlags(f *flag.FlagSet) {
...
-	f.Var(&cfg.ServiceAccount, prefix+"gcs.service-account", "JSON representing either a Google Developers Console client_credentials.json file or a Google Developers service account key file. If empty, fallback to Google default logic.")
+	f.Var(&cfg.ServiceAccount, prefix+"gcs.service-account", "JSON either from a Google Developers Console client_credentials.json file, or a Google Developers service account key. Needs to be valid JSON, not a filesystem path. "+
+		"If empty, fallback to Google default logic: "+
+		"\n1. A JSON file whose path is specified by the GOOGLE_APPLICATION_CREDENTIALS environment variable. For workload identity federation, refer to https://cloud.google.com/iam/docs/how-to#using-workload-identity-federation on how to generate the JSON configuration file for on-prem/non-Google cloud platforms."+
+		"\n2. A JSON file in a location known to the gcloud command-line tool. On Windows, this is %APPDATA%/gcloud/application_default_credentials.json. On other systems, $HOME/.config/gcloud/application_default_credentials.json."+
+		"\n3. On Google App Engine standard first generation runtimes (<= Go 1.9) it uses the appengine.AccessToken function."+
```
Comment (pracucci) [inline pkg/storage/bucket/gcs/config.go:32]: I think we can remove this step. Mimir doesn't run on Google App Engine.

File: pkg/storage/bucket/gcs/config.go [hunk 3]
```diff
@@ -25,5 +25,10 @@ func (cfg *Config) RegisterFlags(f *flag.FlagSet) {
...
 	f.StringVar(&cfg.BucketName, prefix+"gcs.bucket-name", "", "GCS bucket name")
-	f.Var(&cfg.ServiceAccount, prefix+"gcs.service-account", "JSON representing either a Google Developers Console client_credentials.json file or a Google Developers service account key file. If empty, fallback to Google default logic.")
+	f.Var(&cfg.ServiceAccount, prefix+"gcs.service-account", "JSON either from a Google Developers Console client_credentials.json file, or a Google Developers service account key. Needs to be valid JSON, not a filesystem path. "+
+		"If empty, fallback to Google default logic: "+
+		"\n1. A JSON file whose path is specified by the GOOGLE_APPLICATION_CREDENTIALS environment variable. For workload identity federation, refer to https://cloud.google.com/iam/docs/how-to#using-workload-identity-federation on how to generate the JSON configuration file for on-prem/non-Google cloud platforms."+
+		"\n2. A JSON file in a location known to the gcloud command-line tool. On Windows, this is %APPDATA%/gcloud/application_default_credentials.json. On other systems, $HOME/.config/gcloud/application_default_credentials.json."+
```
Comment (pracucci) [inline pkg/storage/bucket/gcs/config.go:31]: I think we can remove Windows. We don't support it.

**Review by pracucci — APPROVED** [review]
Thanks!

**Pull request merged**